swarm_dial.go

```diff
@@ -649,11 +648,18 @@ func (s *Swarm) filterKnownUndialables(p peer.ID, addrs []ma.Multiaddr) []ma.Mul
 		}
 	}
 
-	return addrutil.FilterAddrs(addrs,
-		addrutil.SubtractFilter(ourAddrs...),
+	return ma.FilterAddrs(addrs,
+		func(addr ma.Multiaddr) bool {
```

Comment:
what happened to `SubtractFilter`? It would be nice to have it and use it here instead of duplicating its code.

Reply:
This was the only occurrence of `SubtractFilter`, and the plan was to remove that function. It allocates a `map`, so it will probably be advantageous to replace it by something that doesn't allocate: https://github.com/libp2p/go-addr-util/blob/f3e9655430934836b83ea4bdf38c8599e0b17128/filter.go#L7-L17.

Reply:
ok, fair enough.